threatgrid_connector.py

```diff
@@ -370,24 +405,79 @@ def _check_existing(self, hash):
 
         return False
 
-    def _get_report(self, task_id, is_download_report=False):
+    def _query_finished_tasks(self, param):
+        url = FINISHED_TASKS_URL.format(base_uri=self.threatgrid_base_uri,
+                                        api_key=self.threatgrid_api_key)
+
+        query = param.get(SEARCH_QUERY)
+        from_curr_user = param.get(CURR_USER_KEY)
+        from_user_org = param.get(FROM_USER_ORG_KEY)
+        limit = self._validate_integer(self.threatgrid_action_result, param.get('limit',
+            DEFAULT_LIMIT), "limit", allow_zero=False)
+        artifact = param.get(ARTIFACT_KEY)
+        finished_type = param.get(FINISHED_TYPE, THREATGRID_DONE_STATES[0])
+
+        if finished_type not in THREATGRID_DONE_STATES:
+            return self.threatgrid_action_result.set_status(phantom.APP_ERROR,
+                "Invalid finished type. Supported types: {}".format(THREATGRID_DONE_STATES))
+
+        if artifact and not query:
+            return self.threatgrid_action_result.set_status(phantom.APP_ERROR,
+                "Query required to search for artifacts")
+        elif artifact and artifact not in ARTIFACT_TYPES:
+            return self.threatgrid_action_result.set_status(phantom.APP_ERROR,
+                "Invalid artifact type. Supported types: {}".format(ARTIFACT_TYPES))
+
+        if limit is None:
+            return self.threatgrid_action_result.get_status()
+
+        url += f"&state={finished_type}"
+
+        if query:
+            url += f"&q={query}"
+        if from_curr_user:
+            url += f"&user_only={from_curr_user}"
+        if from_user_org:
+            url += f"&org_only={from_user_org}"
+        if limit:
+            url += f"&limit={limit}"
+
+        try:
+            r = requests.get(url, verify=self.verify, timeout=DEFAULT_TIMEOUT)
+            obj = r.json()
+        except Exception as e:
+            print(e)
```

Comment:
did you mean to print this here or debug_print?

Reply:
Yup fixed 👍